### ISSUE TITLE
Wireless Android Auto crashes every 30 seconds

### ISSUE THREAD

**Issue by supernick100** [GrapheneOS/os-issue-tracker#4011]
When it goes through, it successfully streams audio and Google Maps, but it will promptly close after at least half a minute. My car tablet is a third-party tablet that I bought for my old 2008 model car that only supports wireless Android Auto and Apple Carplay, which we know require special permissions that wired Auto does not warrant. 

I've tried: 
* Giving every permission to Google Play services, Framework, and Android Auto on their respective app configs, as well as running them in the background at all times
* and Apps > Sandboxed Google Play > Android Auto

These where done on a separate profile since they are such a compromise to security.

After said processes, it does run! ... but only for thirty seconds before either crashing or closing, however your see it.

Since it is wireless, it does require having its own wifi network that the Bluetooth phone will automatically connect to upon pairing with the tablet.

So I figured the wifi was unstable. The wifi dictates the connection maybe. The phone still uses its personal data for clarification.

Before switching to Graphene, this was never a problem, so the car tablet is not at fault. For example, I borrowed an old Samsung Galaxy A11 and Auto worked on that non-stop despite being an old, slow phone. The Apple iPhone X I used to have worked for hours as well with Apple Carplay. With my Pixel 9 Pro Fold being on Graphene now, wireless Auto is useless sonce it keeps disconnecting every thirty seconds.

So as a result I've tried continual wifi and Bluetooth scanning. No difference. Then wifi it's not the issue here.

Has this ever been solved, or this a case to solve for the developers? Have I exhausted all my options?

**Comment by de0u** [comment]
Does the information in issue #3944 help?

**Comment by supernick100** [comment]
I'll try that tomorrow.

**Comment by muhomorr** [comment]
Try to connect one more time, then immediately capture Android Auto app log from Settings > Apps > Show all apps > Android Auto > View logs. Send the log file to one of the following places:

Matrix: @muhomorr:grapheneos.org
Discord: muhomorr_ (with underscore at the end)
Email: muhomorr@grapheneos.org

**Comment by supernick100** [comment]
I think you're not meant to be able to update Auto through the Play Store since Graphene uses a sandboxed version. You shouldn't have to update it in that scenario.

It also like I can't change wifi bands on either the phone system settings or the car tablet settings.

**Comment by muhomorr** [comment]
@supernick100 I've looked at the logs that you've sent. This issue is almost certainly due to upstream Bluetooth/Wi-Fi connectivity issues on 9th generation Pixel devices. There are reports of similar Android Auto issues on stock OS online:
https://www.reddit.com/r/Pixel9Pro/comments/1ezwo0x/anyone_have_issues_with_android_auto_so_far/
https://www.reddit.com/r/GooglePixel/comments/1f36rqy/9_pro_xl_android_auto_disconnections/
https://www.androidpolice.com/google-pixel-9-buyers-report-bluetooth-connectivity-issues/
etc

@FricoRico Are you using a 9th generation Pixel device?

**Comment by FricoRico** [comment]
Yes I'm using Pixel 9 Pro

**Comment by FricoRico** [comment]
The issue still persist. I was hoping that Android 15 update would resolve the issues. Has there been any news From Google about the issue?

To be sure, I have attached the logs after connection.

[Android Auto log 7e9ad8ba9d89.txt](https://github.com/user-attachments/files/17598436/Android.Auto.log.7e9ad8ba9d89.txt)

**Comment by sporterconnect** [comment]
Using a Pixel 9 with a MA1 wireless adapter and have no issues with disconnects, just throwing this in here in case it helps narrow things down.

**Comment by Whatsek** [comment]
I have the same issue, also with a Pixel 9 Pro XL and AAWireless, I found this thread googling for answers. Just to confirm that it indeed looks like a Pixel 9 Pro issue. I will test with my Pixel 8 with GOS in the next days, with the same hardware and settings as on my 9 and let you know.

**Comment by muhomorr** [comment]
> Do you perhaps have some more inside knowledge about the current state of the issue?

No. It's almost certainly a software issue. Quarterly Android update (15 QPR1) should be out next week, it's likely to resolve this.

**Comment by FricoRico** [comment]
> > Do you perhaps have some more inside knowledge about the current state of the issue?
> 
> No. It's almost certainly a software issue. Quarterly Android update (15 QPR1) should be out next week, it's likely to resolve this.

I can confirm that update 2024120900 fixes repeated disconnects from wireless Android Auto.

**Comment by way-lo** [comment]
@FricoRico Can you comment if you have had any issues pop back up with wireless AA on your Pixel?

I have a Pixel 9, still on Android 14, that has what may be the same issue.  It will work fine for a couple of trips, but then it will crash, and it is very difficult to get to re-connect.  Reboots of phone and car entertainment center do not reliable restore it.  Wired AA works without any issues.

**Comment by FricoRico** [comment]
> [@FricoRico](https://github.com/FricoRico) Can you comment if you have had any issues pop back up with wireless AA on your Pixel?
> 
> I have a Pixel 9, still on Android 14, that has what may be the same issue. It will work fine for a couple of trips, but then it will crash, and it is very difficult to get to re-connect. Reboots of phone and car entertainment center do not reliable restore it. Wired AA works without any issues.

No issues as of late. I'm running Android 15. When I had issues before it was very rythmic and every 30 seconds or so it would crash. It would never work for a single trip. Your issue sounds different to the issues back then.

**Comment by way-lo** [comment]
Thanks for the information.  It does sound slightly different.  But there do seem to be a lot of reports online re: wireless AA crashing with the PIxel 9, moreso the Pro version.

One more reason to upgrade to the lastest A15.  Unfortunately the bootloader is locked on my P9 and I can only run stock until the carrier unlock expires in 1.5 years.
